tests/autoscaler/pool_manager_test.py

```diff
@@ -239,6 +242,93 @@ def test_successful(self, mock_logger, mock_pool_manager):
+            )]
+        }
+        assert mock_pool_manager._filter_scale_up_options_for_pod(pod1, options) == options
+
+    def test_filter_scale_up_options_unschedulable(self, mock_pool_manager, pod1, caplog):
+        options = {
+            'rg1': [ClusterNodeMetadata(
+                agent=AgentMetadata(
+                    allocated_resources=ClustermanResources(),
+                    unschedulable=True,
+                ),
+                instance=mock.Mock(),
+            )]
+        }
+        assert mock_pool_manager._filter_scale_up_options_for_pod(pod1, options) == {}
+        assert 'node is not schedulable' in caplog.text
+
+    def test_filter_scale_up_options_insufficient_resources(self, mock_pool_manager, pod1, caplog):
+        options = {
+            'rg1': [ClusterNodeMetadata(
+                agent=AgentMetadata(
+                    allocated_resources=ClustermanResources(),
+                ),
+                instance=mock.Mock(),
+            )]
+        }
+        assert mock_pool_manager._filter_scale_up_options_for_pod(pod1, options) == {}
+        assert 'insufficient resources' in caplog.text
+
+    def test_filter_scale_up_options_max_tasks(self, mock_pool_manager, pod1, caplog):
+        options = {
+            'rg1': [ClusterNodeMetadata(
+                agent=AgentMetadata(
+                    allocated_resources=ClustermanResources(cpus=20, mem=1000, disk=1000, gpus=0),
+                    max_tasks=10,
+                    task_count=10,
+                ),
+                instance=mock.Mock(),
+            )]
+        }
+        assert mock_pool_manager._filter_scale_up_options_for_pod(pod1, options) == {}
+        assert 'node running max tasks' in caplog.text
+
+    @pytest.mark.parametrize('node_labels', [{}, {'clusterman.com/pool': 'regular_pool'}])
+    def test_filter_scale_up_options_no_node_selector(self, mock_pool_manager, pod1, node_labels, caplog):
+        pod1.spec.node_selector = {'clusterman.com/pool': 'special_pool'}
+        options = {
+            'rg1': [ClusterNodeMetadata(
+                agent=AgentMetadata(
+                    allocated_resources=ClustermanResources(cpus=20, mem=1000, disk=1000, gpus=0),
+                    labels=node_labels,
+                ),
+                instance=mock.Mock(),
+            )]
+        }
+        assert mock_pool_manager._filter_scale_up_options_for_pod(pod1, options) == {}
+        assert 'mismatched node selector' in caplog.text
+
+    def test_filter_scale_up_options_node_not_tolerable(self, mock_pool_manager, pod1, caplog):
+        options = {
+            'rg1': [ClusterNodeMetadata(
+                agent=AgentMetadata(
+                    allocated_resources=ClustermanResources(cpus=20, mem=1000, disk=1000, gpus=0),
+                ),
+                instance=mock.Mock(),
+            )]
+        }
+        with mock.patch('clusterman.autoscaler.pool_manager.is_node_tolerable', return_value=False):
```

Comment:
nit: I'd prefer if this were a function decorator.

Reply:
I dislike using mock.patch as a function decorator and prefer the context manager, I'm going to discard unless you feel strongly :)